### PR TITLE
[FS] Optmize RemoveTransaction to use the new lookups

### PR DESCRIPTION
**Pull request by quantumagi**

### CODE REVIEW

**Comment by fassadlr** [comment]
@quantumagi this PR looks like it builds on from the previous one?

File: src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs [hunk 1]
```diff
@@ -797,18 +797,38 @@ private bool RewindIfRequired()
                     dbreezeTransaction.ValuesLazyLoadingIsOn = false;
 
                     ChainedHeader prevTip = this.TipHashAndHeight;
+                    int prevDepositHeight = this.NextMatureDepositHeight;
 
                     try
                     {
                         StatusChangeTracker tracker = this.OnDeleteBlocks(dbreezeTransaction, fork.Height);
+
```
Comment (codingupastorm) [inline src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs]: Remove this file from this PR?

Comment (fassadlr) [reply src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs]: @codingupastorm there are changes here from 3690 ;)

**Comment by quantumagi** [comment]
Yeah a bit unfortunate.. This will fix itself once 3690 is committed.

**Comment by fassadlr** [comment]
> @fassadlr , It was branched off 3690. This will fix itself once 3690 is merged.

@quantumagi I know but we normally don't let PRs follow on from each other like this because if there any changes to 3690 it will affect this PR as well.. .which is not good. If 3691 doesn't depend on 3690 then redo the changes in a new PR. If it depends on it then just wait for 3690 to be merged before creating this one.

**Comment by quantumagi** [comment]
Replaced by #3692.

**Pull request closed without merging**